apps/services/user-profile/src/app/v2/user-profile.service.ts

```diff
@@ -36,6 +49,175 @@ export class UserProfileService {
       mobilePhoneNumberVerified: userProfile.mobilePhoneNumberVerified,
       emailVerified: userProfile.emailVerified,
       documentNotifications: userProfile.documentNotifications,
+      needsNudge: this.checkNeedsNudge(userProfile.lastNudge as Date),
+    }
+  }
+
+  async patch(
+    nationalId: string,
+    userProfile: PatchUserProfileDto,
+  ): Promise<UserProfileDto> {
+    const isEmailDefined = isDefined(userProfile.email)
+    const isMobilePhoneNumberDefined = isDefined(userProfile.mobilePhoneNumber)
+
+    if (
+      isEmailDefined &&
+      userProfile.email !== '' &&
+      !isDefined(userProfile.emailVerificationCode)
+    ) {
+      throw new BadRequestException('Email verification code is required')
+    }
+
+    if (
+      isMobilePhoneNumberDefined &&
+      userProfile.mobilePhoneNumber !== '' &&
+      !isDefined(userProfile.mobilePhoneNumberVerificationCode)
+    ) {
+      throw new BadRequestException(
+        'Mobile phone number verification code is required',
+      )
+    }
+
+    const { nationalNumber, parsedPhoneNumber } =
+      isMobilePhoneNumberDefined &&
+      userProfile.mobilePhoneNumber !== '' &&
+      this.formatPhoneNumber(userProfile.mobilePhoneNumber)
+
+    return await this.sequelize.transaction(async (transaction) => {
+      if (userProfile.email !== '') {
+        isEmailDefined &&
+          (await this.verificationService
+            .confirmEmail(
+              {
+                email: userProfile.email,
+                hash: userProfile.emailVerificationCode,
+              },
+              nationalId,
+              transaction,
+            )
+            .then((res) => {
+              if (!res.confirmed) {
+                throw new BadRequestException(res.message)
+              }
+            }))
+      }
+
+      if (userProfile.mobilePhoneNumber !== '') {
+        isMobilePhoneNumberDefined &&
+          (await this.verificationService
+            .confirmSms(
+              {
+                mobilePhoneNumber: userProfile.mobilePhoneNumber,
+                code: userProfile.mobilePhoneNumberVerificationCode,
+              },
+              nationalId,
+              transaction,
+            )
+            .then((res) => {
+              if (!res.confirmed) {
+                throw new BadRequestException(res.message)
+              }
+            }))
+      }
```

Comment:
You could simplify it to something of the following

```suggestion
      const commonArgs = [nationalId, transaction] as const
      const promiseResults = await Promise.all(
        [
          isEmailDefined &&
            this.verificationService.confirmEmail(
              {
                email: userProfile.email,
                hash: userProfile.emailVerificationCode,
              },
              ...commonArgs,
            ),
          isMobilePhoneNumberDefined &&
            this.verificationService.confirmSms(
              {
                mobilePhoneNumber: userProfile.mobilePhoneNumber,
                code: userProfile.mobilePhoneNumberVerificationCode,
              },
              ...commonArgs,
            ),
        ].filter(Boolean),
      )

      promiseResults.forEach(({ confirmed, message }) => {
        if (!confirmed) {
          throw new BadRequestException(message)
        }
      })
```